### PR TITLE
Add JsonAdaptedResumeTest

### PR DESCRIPTION
**Pull request by duongphammmm**

### CODE REVIEW

File: src/test/java/seedu/address/storage/JsonAdaptedResumeTest.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        jsonAdaptedResume = new JsonAdaptedResume(resumeWithTags);
+        assertEquals(resumeWithTags, jsonAdaptedResume.toModelType());
+
+        jsonAdaptedResume = new JsonAdaptedResume(VALID_RESUME_NAME_ME, "4",
+                internshipIds, projectIds, skillIds, tags);
+        assertEquals(resumeWithTags, jsonAdaptedResume.toModelType());
+    }
+
+    @Test
+    public void loadToModel_invalidName_failure() {
```
Comment (duongphammmm) [inline src/test/java/seedu/address/storage/JsonAdaptedResumeTest.java]:
```suggestion
    public void toModelType_invalidName_throwsIllegalValueException() {
```

File: src/test/java/seedu/address/storage/JsonAdaptedResumeTest.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        JsonAdaptedResume jsonAdaptedResume = new JsonAdaptedResume(INVALID_NAME_DESC, "4",
+                internshipIds, projectIds, skillIds, tags);
+        assertThrows(IllegalValueException.class, Name.MESSAGE_CONSTRAINTS, jsonAdaptedResume::toModelType);
+    }
+
+    @Test
+    public void loadToModel_missingName_failure() {
```
Comment (duongphammmm) [inline src/test/java/seedu/address/storage/JsonAdaptedResumeTest.java]:
```suggestion
    public void toModelType_missingName_throwsIllegalValueException() {
```

File: src/test/java/seedu/address/storage/JsonAdaptedResumeTest.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        JsonAdaptedResume jsonAdaptedResume = new JsonAdaptedResume(null, "4",
+                internshipIds, projectIds, skillIds, tags);
+        assertThrows(IllegalValueException.class,
+                String.format(MISSING_FIELD_MESSAGE_FORMAT, Name.class.getSimpleName()),
+                jsonAdaptedResume::toModelType);
+    }
+
+    @Test
+    public void loadToModel_invalidId_failure() {
```
Comment (duongphammmm) [inline src/test/java/seedu/address/storage/JsonAdaptedResumeTest.java]:
```suggestion
    public void toModelType_invalidId_throwsIllegalValueException() {
```

File: src/test/java/seedu/address/storage/JsonAdaptedResumeTest.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+                jsonAdaptedResume::toModelType);
+
+        jsonAdaptedResume = new JsonAdaptedResume(VALID_RESUME_NAME_ME, "-4",
+                internshipIds, projectIds, skillIds, tags);
+        assertThrows(IllegalValueException.class, "The id field must not be negative.",
+                jsonAdaptedResume::toModelType);
+    }
+
+    @Test
+    public void loadToModel_missingId_failure() {
```
Comment (duongphammmm) [inline src/test/java/seedu/address/storage/JsonAdaptedResumeTest.java]:
```suggestion
    public void toModelType_missingId_throwsIllegalValueException() {
```

File: src/test/java/seedu/address/storage/JsonAdaptedResumeTest.java [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+        JsonAdaptedResume jsonAdaptedResume = new JsonAdaptedResume(VALID_RESUME_NAME_ME, null,
+                internshipIds, projectIds, skillIds, tags);
+        assertThrows(IllegalValueException.class,
+                String.format(MISSING_FIELD_MESSAGE_FORMAT, "Id"), jsonAdaptedResume::toModelType);
+    }
+
+    @Test
+    public void loadToModel_invalidItemIds_failure() {
```
Comment (duongphammmm) [inline src/test/java/seedu/address/storage/JsonAdaptedResumeTest.java]:
```suggestion
    public void toModelType_invalidItemIds_throwsIllegalValueException() {
```

File: src/main/java/seedu/address/storage/JsonAdaptedResume.java [hunk 6]
```diff
@@ -85,34 +85,44 @@ public Resume toModelType() throws IllegalValueException {
...
 
-        final int modelId;
-        try {
-            modelId = Integer.parseInt(id);
-        } catch (NumberFormatException e) {
-            throw new IllegalValueException("The id field can only be an integer.");
-        }
-        if (modelId < 0) {
-            throw new IllegalValueException("The id field must not be negative.");
-        }
```
Comment (wardetu) [inline src/main/java/seedu/address/storage/JsonAdaptedResume.java:96]: Why is this portion removed ?

Comment (helloImHai) [reply src/main/java/seedu/address/storage/JsonAdaptedResume.java:96]: She abstracted it to a parseId method

**Review by helloImHai — APPROVED** [review]
Clean and I love it 🛀🛀🛀
Can merge

**Pull request merged**